packages/roosterjs-editor-dom/lib/inlineElements/resolveInlineElement.ts

```diff
@@ -22,19 +22,19 @@ export default function resolveInlineElement(
         parent = parent.parentNode
     ) {
         nodeChain.push(parent);
-        parent = parent.parentNode;
     }
 
     let inlineElement: InlineElement;
 
     for (let i = nodeChain.length - 1; i >= 0 && !inlineElement; i--) {
-        let tag = getTagOfNode(nodeChain[i]);
+        let currentNode = nodeChain[i];
+        let tag = getTagOfNode(currentNode);
         if (tag == 'A') {
-            inlineElement = new LinkInlineElement(node, parentBlock);
```

Comment:
Did this just not work as intended before?

Reply:
It was just introduced from my refactoring, so before 6.10.9 it worked well.